src/main/scala/com/fulcrumgenomics/umi/VanillaUmiConsensusCaller.scala

```diff
@@ -196,39 +205,57 @@ class VanillaUmiConsensusCaller(override val readNamePrefix: String,
       val consensusDepths = new Array[Short](consensusLength)
       val consensusErrors = new Array[Short](consensusLength)
 
-      var positionInRead = 0
-      val builder = this.caller.builder()
-      while (positionInRead < consensusLength) {
-        // Add the evidence from all reads that are long enough to cover this base
-        capped.foreach { read =>
-          if (read.length > positionInRead) {
-            val base = read.bases(positionInRead)
-            val qual = read.quals(positionInRead)
-            if (base != NoCall) builder.add(base=base, qual=qual)
-          }
-        }
+      if (capped.length == 1) {
+        val inBases = capped.head.bases
+        val inQuals = capped.head.quals
 
-        val depth = builder.contributions // NB: cache this value, as it is re-computed each time
+        forloop (from=0, until=consensusLength) { i =>
+          val rawBase     = inBases(i)
+          val rawQual     = SingleInputConsensusQuals(inQuals(i))
+          val (base, qual) = if (rawQual < this.options.minConsensusBaseQuality) (NoCall, TooLowQualityQual) else (rawBase, rawQual)
+          val isNoCall = base == NoCall
 
-        // Call the consensus and do any additional filtering
-        val (rawBase, rawQual) = builder.call()
-        val (base, qual) = {
-          if (depth < this.options.minReads)       (NoCall, NotEnoughReadsQual)
-          else if (rawQual < this.options.minConsensusBaseQuality) (NoCall, TooLowQualityQual)
-          else (rawBase, rawQual)
+          consensusBases(i)  = base
+          consensusQuals(i)  = qual
+          consensusDepths(i) = if (isNoCall) 0 else 1
+          consensusErrors(i) = 0
         }
+      }
+      else {
```

Comment:
Everything from here on down is identical, just indented under an else, and github has gotten confused.